sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Utils.java

```diff
@@ -741,4 +734,15 @@ public static String toJson(ObjectMapper mapper, ObjectNode object) {
         os.writeObject(obj);
         return out.toByteArray();
     }
+
+    public static long getMaxIntegratedCacheStalenessInMillis(DedicatedGatewayRequestOptions dedicatedGatewayRequestOptions) {
```

Comment:
Is there a reason to not move all the logic to the helper?

```java
public static void addMaxIntegratedCacheStalenessInMillis(
    Headers headers,
    DedicatedGatewayRequestOptions dedicatedGatewayRequestOptions){

     if (dedicatedGatewayRequestOptions == null ||
          dedicatedGatewayRequestOptions.getMaxIntegratedCacheStaleness() == null){
         return;
      }

      Duration maxIntegratedCacheStaleness = dedicatedGatewayRequestOptions.getMaxIntegratedCacheStaleness();
        if (maxIntegratedCacheStaleness.toNanos() > 0 && maxIntegratedCacheStaleness.toMillis() <= 0) {
            throw new IllegalArgumentException("MaxIntegratedCacheStaleness granularity is milliseconds");
        }
        if (maxIntegratedCacheStaleness.toMillis() < 0) {
            throw new IllegalArgumentException("MaxIntegratedCacheStaleness duration cannot be negative");
        }
        headers.put(HttpConstants.HttpHeaders.DEDICATED_GATEWAY_PER_REQUEST_CACHE_STALENESS,
           String.valueOf(maxIntegratedCacheStaleness));
}

// Caller just needs this:
Utils.addMaxIntegratedCacheStalenessInMillis(headers, options.getDedicatedGatewayRequestOptions());
```

Reply:
Discussed offline, for others, I will just put my comments here.
 
As you notice, all the header values are being added in the same file and in the same way. So when someone comes and reads the code, they won't have to figure out what header value we are adding and under what name. It will be very clear from the code, that this is the header name and header value. If they want to go deep, they can go into the method to see how we are retrieving the value. 
This is purely for code readability and quick debugging purpose, no other intention.